pandas/core/categorical.py

```diff
@@ -1364,6 +1365,54 @@ def sort_values(self, inplace=False, ascending=True, na_position='last'):
             return self._constructor(values=codes, categories=self.categories,
                                      ordered=self.ordered, fastpath=True)
 
+    def rank(self, method='average', na_option='keep',
+             ascending=True, pct=False):
```

Comment:
instead of duplicating the doc-string. Move the one from ``pandas/core/generic.py`` to ``pandas/core/base.py`` (ONLY the doc-string) and put it in shared_docs. Then import to Categorical as well.